### ISSUE TITLE
Windows: npx is not escaping paths correctly

### ISSUE THREAD

**Issue by zkat** [zkat/npx#84]
Ref: https://github.com/zkat/npx/issues/62#issuecomment-315564704

I'm confused 'cause this should totally be the right way to escape things for `child_process.exec`. Note that `C:\"Program Files"\...` is there, which _should_ be the way to escape paths on Windows.

I'm thinking it's possible that this sort of escaping _only_ applies when you're dealing with the binary path itself, and arguments _don't_ get processed the same way and thus need to be escaped as args?

**Comment by katemihalikova** [comment]
Right. What if `npx` just quotes whole args and all of them? `"C:\Program Files\nodejs\node.exe" "C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js" "-v"` works just fine in `cmd.exe`.

**Comment by zkat** [comment]
I think that's what needs to be done, yeah.

**Comment by simonua** [comment]
Thanks a lot for keeping on this! Pointing out the obvious but just so you know for certain what my setup looks like:

Node is located at **C:\Program Files\nodejs\node.exe** (spaces in path)
npm is located at **C:\Users\\[userid]\AppData\Roaming\npm\npm.cmd** (no spaces in path)

**Comment by zkat** [comment]
Let's see if https://github.com/zkat/npx/pull/87 works now?

**Comment by katemihalikova** [comment]
It's working fine for me. Works in both cmd.exe and git bash.

Since my `npm` path doesn't contain spaces, I've tried with `npx --npm="C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js" cowsay hi` and it's also working fine.

**Comment by simonua** [comment]
Nice! =) Thank you kindly, @zkat and @katemihalikova!

![image](https://user-images.githubusercontent.com/8005145/28294840-9a0e4912-6b2a-11e7-88c5-452c8f7f3702.png)

**Comment by simonua** [comment]
@zkat, sorry, quick note for the back of your mind: Looks like it should say "installed 5 **packages** in 21.205s.", yes?

**Comment by zkat** [comment]
lol probably, but adding a single word to that string involves updating 17 other translations, many to languages I don't even know how to touch 😂

**Comment by simonua** [comment]
And pluralization. =P

**Comment by aivo0** [comment]
This issue is still in Webpack version 4.17.1. Why is it closed?

**Comment by scambier** [comment]
Same problem here
```
> npx -v
Command failed: C:\"Program Files"\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js config get cache --parseable
module.js:491
    throw err;
    ^

Error: Cannot find module 'C:\Program'
    at Function.Module._resolveFilename (module.js:489:15)
    at Function.Module._load (module.js:439:25)
    at Function.Module.runMain (module.js:609:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:598:3

9.2.1
```

**Comment by scambier** [comment]
@aivo0 for information, I solved it by  upgrading to node v8.12.0
